CMakeLists.txt

```diff
@@ -18,6 +18,18 @@ option(DOWNLOAD_ENSMALLEN "If ensmallen is not found, download it." ON)
 option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_PYTHON_BINDINGS "Build Python bindings." ON)
 option(BUILD_JULIA_BINDINGS "Build Julia bindings." ON)
+set(BOARD_NAME "" CACHE STRING "Enter the board name to compile mlpack for.")
+set(ARMADILLO_INCLUDE_DIR "" CACHE STRING "Path for Armadillo headers.")
+set(OPENBLAS_LIBRARIES "" CACHE STRING "Path for OpenBlas library.")
+set(BOOST_INCLUDE_DIR "" CACHE STRING "Path for boost headers.")
+
+if (CMAKE_CROSSCOMPILING AND (BOARD_NAME STREQUAL ""))
+  message(FATAL_ERROR "Please enter the board name")
+elseif(CMAKE_CROSSCOMPILING AND (BOARD_NAME STRGREATER ""))
+  message(STATUS "Using toolchain file: ${CMAKE_TOOLCHAIN_FILE}.")
+  include(board/flags-config.cmake)
+  flags(${BOARD_NAME})
+endif()
 
 if (WIN32)
   option(BUILD_SHARED_LIBS
```

Comment:
I don't think Github will let me comment at exactly the right place, but we can now remove `program_options` from the link dependencies throughout the CMake configuration.